diff --git a/Makefile b/Makefile
--- ./Makefile
+++ ./Makefile
@@ -44,6 +44,8 @@ define build_package
 $1-$2:
 	cargo build --release --features=$1 -p avante-$2
 	cp target/release/libavante_$(shell echo $2 | tr - _).$(EXT) $(BUILD_DIR)/avante_$(shell echo $2 | tr - _).$(EXT)
+	@mkdir -p lua/avante
+	cp $(BUILD_DIR)/avante_$(shell echo $2 | tr - _).$(EXT) lua/avante/
 endef
 
 define build_targets
diff --git a/lua/avante/config.lua b/lua/avante/config.lua
--- ./lua/avante/config.lua
+++ ./lua/avante/config.lua
@@ -47,7 +47,7 @@ M.defaults = {
   ---@type AvanteSupportedProvider
   claude = {
       endpoint = "https://api.anthropic.com",
-    model = "claude-3-5-sonnet-20241022",
+      model = "claude-3-5-haiku-20241022",
       timeout = 30000, -- Timeout in milliseconds
       temperature = 0,
       max_tokens = 8000,
diff --git a/lua/avante/providers/claude.lua b/lua/avante/providers/claude.lua
--- ./lua/avante/providers/claude.lua
+++ ./lua/avante/providers/claude.lua
@@ -1,6 +1,7 @@
 local Utils = require("avante.utils")
 local Clipboard = require("avante.clipboard")
 local P = require("avante.providers")
+local Logger = require("avante.logger")
 
 ---@class AvanteClaudeBaseMessage
 ---@field cache_control {type: "ephemeral"}?
@@ -106,12 +107,8 @@ M.parse_curl_args = function(provider, prompt_opts)
 
   local messages = M.parse_messages(prompt_opts)
 
-  return {
-    url = Utils.url_join(base.endpoint, "/v1/messages"),
-    proxy = base.proxy,
-    insecure = base.allow_insecure,
-    headers = headers,
-    body = vim.tbl_deep_extend("force", {
+  local url = Utils.trim(base.endpoint, { suffix = "/" }) .. "/v1/messages"
+  local body = vim.tbl_deep_extend("force", {
       model = base.model,
       system = {
         {
@@ -122,11 +119,22 @@ M.parse_curl_args = function(provider, prompt_opts)
       },
       messages = messages,
       stream = true,
-    }, body_opts),
+  }, body_opts)
+
+  Logger.debug_request(url, headers, body)
+
+  return {
+    url = url,
+    proxy = base.proxy,
+    insecure = base.allow_insecure,
+    headers = headers,
+    body = body,
   }
 end
 
 M.on_error = function(result)
+  Logger.debug_response(result)
+
   if not result.body then
     return Utils.error("API request failed with status " .. result.status, { once = true, title = "Avante" })
   end
diff --git a/lua/avante/sidebar.lua b/lua/avante/sidebar.lua
--- ./lua/avante/sidebar.lua
+++ ./lua/avante/sidebar.lua
@@ -1720,16 +1720,6 @@ function Sidebar:render(opts)
     xpcall(function() api.nvim_buf_set_name(self.result.bufnr, RESULT_BUF_NAME) end, function(_) end)
   end)
 
-  self.result:map("n", "q", function()
-    Llm.cancel_inflight_request()
-    self:close()
-  end)
-
-  self.result:map("n", "<Esc>", function()
-    Llm.cancel_inflight_request()
-    self:close()
-  end)
-
   self:create_input(opts)
 
   self:update_content_with_history(chat_history)
